localstack/aws/handlers/service_plugin.py

```diff
@@ -47,24 +47,42 @@ def require_service(self, _: HandlerChain, context: RequestContext, response: Re
                 f"Service '{service_name}' is not enabled. Please check your 'SERVICES' configuration variable."
             )
 
-        service_operation: Optional[ServiceOperation] = context.service_operation
+        if service_name in self.loaded_services:
+            return
```

Comment:
nit: I see that we can save a few operations if we keep the state of loaded services here, but I feel like this should be in the responsibility of the `service _manager`. But also this code is executed on every single request, so it might make sense to avoid the plugin handling logic if possible.